service/controller/resource/restrictawsnodedaemonset/create.go

```diff
@@ -134,3 +114,39 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	return nil
 }
+
+func ensureAndFilterNodeSelectorRequirements(requirements []corev1.NodeSelectorRequirement) ([]corev1.NodeSelectorRequirement, bool) {
+	ret := make([]corev1.NodeSelectorRequirement, 0)
+	changed := false
+	found := false
+
+	for _, requirement := range requirements {
+		// Check if Key or Operator are not the ones we're looking for.
+		if requirement.Key != label.OperatorVersion || requirement.Operator != "NotIn" {
+			// This requirement is acting on a different label than the one we care, keep it as-is.
+			ret = append(ret, requirement)
+		} else {
+			// This requirement is using the same key and operator as the one we expected to set, so let's check if the value matches.
+			if len(requirement.Values) != 1 || requirement.Values[0] != project.Version() {
```

Comment:
would it make more sense to have `len(requirement.Values) > 0`?

Reply:
that is not what we want. When the `key` and `operator` fields match, we want the value to be just one and with a specific value.
What this If is doing, is recognize when there is something off, for example:

- more than one value => `len(..)` != 1`
- wrong value

in those 2 cases we want to replace everything with our only desired value.
does it make sense?

Reply:
Got it, thanks for the explanation. Maybe worth adding this explanation as comment to the function?

Reply:
:done: